### PR TITLE
Add mark as delivered button in chat

### PR DESCRIPTION
**Pull request by jamessspanggg**
- New field `status` added to the chats documents, currently can only be of `delivered` status.
- Badge will be shown for chat with posts that are completed/closed, or delivered. (in the display order of completed/closed -> delivered)
- The mark as delivered button will only be shown from the donor's perspective, when chatting for a wish
- Reword `Complete` to `Complete Wish`/`Complete Donation` respectively

Sample screenshot: LHS(donor), RHS(NPO)
Screen Shot 2020-10-11 at 8 38 57 PM

### CODE REVIEW

File: src/components/modal/ConfirmDeliveredModal.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  setIsNewChat,
+  onShow,
+  onClose,
+}) => {
+  if (!onShow) {
+    return null;
+  }
+  const router = useRouter();
+  const deliveredMessage =
+    'I have delivered the items to the beneficiary, please confirm and mark the wish as completed!';
```
Comment (jamessspanggg) [inline src/components/modal/ConfirmDeliveredModal.js:22]: Please check if the following deliveredMessage is appropriate

File: utils/api/chats.js [hunk 2]
```diff
@@ -426,6 +426,26 @@ class ChatsAPI {
...
+  async markWishAsDelivered(id) {
+    let chatDoc = await this.getChat(id);
+    if (!chatDoc.exists) {
+      throw new ChatError('invalid-chat', `chat of ${id} does not exist`);
+    }
+    let data = {
+      status: DELIVERED,
+    };
+    chatsCollection.doc(id).update(data);
+    chatDoc = await this.getChat(id);
```
Comment (jinyingtan) [inline utils/api/chats.js:445]: Can just return `this.getChat(id)` as most of the api return after `.get()`

**Review by jinyingtan — APPROVED** [review]
LGTM on API changes

**Review by kohchihao — CHANGES_REQUESTED** [review]
LGTM but need to update the colors to new format and fix certain merge conflict.

**Review by kohchihao — APPROVED** [review]
LGTM 👍

**Pull request merged**